src/plugins/external_tool_support/src/ExternalToolManager.cpp

```diff
@@ -193,28 +190,36 @@ QString ExternalToolManagerImpl::addTool(ExternalTool *tool) {
         foreach (const QString &dependency, toolDependencies) {
             dependencies.insertMulti(dependency, tool->getId());
         }
+    }
+}
 
-        if (dependenciesAreOk(tool->getId()) && !tool->isValid()) {
-            if (tool->isModule()) {
-                QString masterId = tool->getDependencies().first();
+QString ExternalToolManagerImpl::addToolToListsAndReturnToolPath(const ExternalTool &tool) {
+    QString toolPath;
+    SAFE_POINT(nullptr != etRegistry, "The external tool registry is nullptr", toolPath);
+
+    QStringList toolDependencies = tool.getDependencies();
+    if (!toolDependencies.isEmpty()) {
+        if (dependenciesAreOk(tool.getId()) && !tool.isValid()) {
+            if (tool.isModule()) {
+                QString masterId = tool.getDependencies().first();
                 ExternalTool *masterTool = etRegistry->getById(masterId);
-                SAFE_POINT(masterTool, QString("An external tool '%1' isn't found in the registry").arg(masterId), "");
+                SAFE_POINT(nullptr != masterTool, QString("An external tool '%1' isn't found in the registry").arg(masterId), toolPath);
```

Comment:
Do not use Yoda style checks in UGENE.
This style makes your code less readable and does not add extra safety: assignments in expression will break UGENE build.

Reply:
Better build failure than unexpected behavior

Reply:
Assignment in expression won't cause build failure. Only a warning will be added that could be skipped easily. Yoda style adds some extra safety.

Reply:
I'm not sure what will happen on Windows, but the code like 
`if (a=2) `
will break Linux build and we will see it in the Teamcity
I think there is the same option for the Windows compiler.

Reply:
I've checked it on macOS with clang, there was a warning, but UGENE was built successfully. If you want to treat this case as error feel free to edit ugene_global.pro appropriately. Until then Yoda style is acceptable. At the same time, I won't insist on using it everywhere.
Also, I can't say that non-Yoda styled code is more readable. I just can't see any difference.

Reply:
Ok, agreed, lets leave it.